src/main/java/org/folio/rest/support/EffectiveCallNumberComponentsUtil.java

```diff
@@ -36,9 +39,17 @@ private static EffectiveCallNumberComponents updateComponents(
       updatedCallNumberSuffix = holdings.getCallNumberSuffix();
     }
 
+    String updatedCallNumberTypeId = StringUtils.firstNonBlank(
+      item.getItemLevelCallNumberTypeId(),
+      Optional.ofNullable(holdings).map(HoldingsRecord::getCallNumberTypeId)
+        .orElse(null)
```

Comment:
I think we may be abusing `Optional`, I know I am guilty of it as well and I need to stop, but this is better suited for a ternary operator. `Optional` is really meant for return types as described here (from an Oracle blog):
> The purpose of `Optional` is not to replace every single null reference in your codebase but rather to help design better APIs in which—just by reading the signature of a method—users can tell whether to expect an optional value.
```suggestion
        holdings != null ? holdings.getCallNumberTypeId() : null
```